demisto_sdk/commands/validate/validate_manager.py

```diff
@@ -1715,3 +1730,46 @@ def check_and_validate_deprecated(self, file_type, file_path, current_file, is_m
 
             return is_valid_as_deprecated
         return None
+
+    def validate_alternative_fields(self, item_data: dict, file_type: FileType):
+        r"""
+            Validate the use of alternative fields related to this item. this inclueds:
+            1. Check if the given item is using any item that has alternative fields, and those are missing from
+             its data, using the id set.
+            2. Check if this current item has an alternative field, and if sp check if all items that are using this
+            current item have its alternative field in their file too.
+        Args:
+            item_data: The extracted data of the item from yml\json.
+            file_type: The type of content item the data belongs to.
+
+        Returns:
+            False if the file is invalid due to missing alternative fields, True otherwise.
+
+        """
+        is_valid = True
+
+        # validate the use of alternative fields in current item's file
+        if check_and_add_missing_alternative_fields(item_data, file_type, self.id_set_file):
+            error_message, error_code = Errors.missing_alternative_fields_nested_items()
+            self.handle_error(error_message, error_code, file_path=self.file_path,
+                              suggested_fix=Errors.suggest_fix(self.file_path, '-s', self.id_set_path))
+            is_valid = False
+
+        # validate alternative fields in items that are using this current item
+        item_id = get_id_from_item_data(item_data)
+        alternative_id, alternative_name = get_alternative_id_and_name_from_id_set(item_id, file_type, self.id_set_file)
+        if alternative_id or alternative_name:
+            items_using_current_item_paths = get_all_using_paths(item_id, file_type, self.id_set_file)
+            for path in items_using_current_item_paths:
+                with open(path, 'r') as file:
+                    using_data = json.loads(file.read()) if path.endswith('json') else yaml.load(file)
+                    file_type = find_type(path)
+                    if check_and_add_missing_alternative_fields(using_data, file_type, self.id_set_file):
+                        # TODO: in the above condition function, there is duplicity with checking if the items have an
+                        #  alternative field (since we got here cause we already checked that...)
```

Comment:
@yaakovi for now i left it as is - with the duplicity- since it is working properly and this PR has been dragged for a while. I can edit it to be more efficient but it will require some more changes  and effort. what do you think?